assets/js/types/type-defs/contexts.ts

```diff
@@ -13,6 +13,12 @@ export type ValidationData = {
 	) => void;
 };
 
+export type ProductQueryContext = {
+	query: object;
+	queryId: number;
+	productId: number;
```

Comment:
Are those always available in the context or sometimes undefined?

Reply:
Addressed in ad2b4decd.

Note: we don't need the `productId`, as it is passed in via the `attributes`. So I've removed it in that commit as well.